staging/kos/Makefile

```diff
@@ -46,11 +47,73 @@ deploy/main/50-rs-ipam.yaml: deploy.m4/main/50-rs-ipam.yaml.m4
+	openssl req -x509 -new -nodes -key etcd-tls/certs/ca-key.pem -days 10000 \
+	-out etcd-tls/certs/ca.pem -subj "/CN=kos-etcd-ca"
+	echo 01 > etcd-tls/certs/ca.srl
+
+etcd-tls/certs/peer.crt: etcd-tls/certs/ca.pem
+	# Generate etcd peer tls key + cert req
+	openssl genrsa -out etcd-tls/certs/peer.key 4096
+	openssl req -new -key etcd-tls/certs/peer.key \
+		-out etcd-tls/certs/peer.csr -subj "/CN=kos-etcd-peer" \
+		-config etcd-tls/cnf/peer.cnf
+
+	# Generate etcd peer tls cert without password
+	openssl x509 -req -in etcd-tls/certs/peer.csr \
+		-CA etcd-tls/certs/ca.pem -CAserial etcd-tls/certs/ca.srl \
+		-CAkey etcd-tls/certs/ca-key.pem -days 1500 -extensions v3_req \
+		-out etcd-tls/certs/peer.crt \
+		-extfile etcd-tls/cnf/peer.cnf
+
+	# Give the ca cert the name the etcd operator expects for peer tls
+	cp etcd-tls/certs/ca.pem etcd-tls/certs/peer-ca.crt
+
+etcd-tls/certs/server.crt: etcd-tls/certs/ca.pem
+	# Generate etcd server tls key + cert req
+	openssl genrsa -out etcd-tls/certs/server.key 4096
+	openssl req -new -key etcd-tls/certs/server.key \
+		-out etcd-tls/certs/server.csr -subj "/CN=kos-etcd-server" \
+		-config etcd-tls/cnf/server.cnf
+
+	# Generate etcd server tls cert without password
+	openssl x509 -req -in etcd-tls/certs/server.csr \
+		-CA etcd-tls/certs/ca.pem -CAserial etcd-tls/certs/ca.srl \
+		-CAkey etcd-tls/certs/ca-key.pem -days 1500 -extensions v3_req \
+		-out etcd-tls/certs/server.crt \
+		-extfile etcd-tls/cnf/server.cnf
+
+	# Give the ca cert the name the etcd operator expects for server tls
+	cp etcd-tls/certs/ca.pem etcd-tls/certs/server-ca.crt
+
+etcd-tls/certs/etcd-client.crt: etcd-tls/certs/ca.pem
+	# Generate etcd client tls key + cert req
+	openssl genrsa -out etcd-tls/certs/etcd-client.key 4096
+	openssl req -new -key etcd-tls/certs/etcd-client.key \
+		-out etcd-tls/certs/etcd-client.csr -subj "/CN=kos-etcd-client" \
+		-config etcd-tls/cnf/etcd-client.cnf
+
+	# Generate etcd client tls cert without password
+	openssl x509 -req -in etcd-tls/certs/etcd-client.csr \
+		-CA etcd-tls/certs/ca.pem -CAserial etcd-tls/certs/ca.srl \
+		-CAkey etcd-tls/certs/ca-key.pem -days 1500 -extensions v3_req \
+		-out etcd-tls/certs/etcd-client.crt \
+		-extfile etcd-tls/cnf/etcd-client.cnf
+
+	# Give the ca cert the name the etcd operator expects for client tls
+	cp etcd-tls/certs/ca.pem etcd-tls/certs/etcd-client-ca.crt
+
 .PHONY: deploy
-deploy: deploy/main/50-rs-xs.yaml deploy/main/50-rs-ipam.yaml deploy/main/50-ds-ca.yaml
+deploy: deploy/main/50-rs-xs.yaml deploy/main/50-rs-ipam.yaml deploy/main/50-ds-ca.yaml etcd-tls/certs/peer.crt etcd-tls/certs/server.crt etcd-tls/certs/etcd-client.crt
 	kubectl apply -f deploy/ns && \
 	kubectl apply -f deploy/etcd-operator-rbac && \
 	kubectl apply -f deploy/etcd-operator && \
+	kubectl -n example-com create secret generic etcd-peer-tls --from-file=etcd-tls/certs/peer-ca.crt --from-file=etcd-tls/certs/peer.crt --from-file=etcd-tls/certs/peer.key
```

Comment:
Ah, this is bad; `make deploy` should be idempotent.  So we should go to having the makefile create YAML.  It's pretty easy, the `base64` command is your friend.

Reply:
How does creating the yaml (which I'm in favor of for uniformity's sake) makes a difference wrt idempotency?

Reply:
Aside from "phony" junk, `make foo` is idempotent.  In particular, we could write something like the following in the Makefile.
```
~/.kos/etcd-tls/peer-ca.crt:
    ${commands to create that file}
```